core/src/main/scala/com/raphtory/api/analysis/graphstate/GraphStateImplementation.scala

```diff
@@ -43,6 +43,60 @@ private object AccumulatorImplementation {
 
 import scala.math.Numeric.Implicits.infixNumericOps
 
+private class CounterImplementation[T] () extends Counter[T] {
+  var totalCount : Int = 0
+  var counts: mutable.Map[T, Int] = mutable.Map[T,Int]()
+
+  override def getCounts: mutable.Map[T, Int] = counts
+  override def largest : (T,Int) = counts.maxBy(_._2)
+  override def largest(k: Int): List[(T,Int)] = counts.toList.sortBy(_._2).takeRight(k.min(counts.size))
+
+  def +=(newValue: (T, Int)): Unit = this.synchronized{
```

Comment:
are you 100% sure this will be used across threads? if yes then perhaps using a [**ConcurrentHashMap.compute**](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#compute-K-java.util.function.BiFunction-) would workout better

Reply:
thanks! I'm not totally sure but I'm just following the pattern of the other accumulators there which require it